src/core/control/XournalMain.cpp

```diff
@@ -615,6 +615,7 @@ auto XournalMain::run(int argc, char** argv) -> int {
 
     XournalMainPrivate app_data;
     GtkApplication* app = gtk_application_new("com.github.xournalpp.xournalpp", APP_FLAGS);
+    g_set_prgname("com.github.xournalpp.xournalpp");
```

Comment:
I thought this is called by `gtk_application_new`. 
And since `GLib 2.72, this function can be called multiple times and is fully thread safe. Prior to GLib 2.72, this function could only be called once per process.` Therefore, I would remove it. Can you test if this makes a difference?

Reply:
This line is precsely what fixes the titlebar icon... Removing it makes this PR pointless.
You can test with `g_get_prgname()` that `gtk_application_new()` actually doesn't set the name.

Reply:
If we are a bit scared, we can always test against the GLib version and only fix the bug for up-to-date setups (i.e. glib >= 2.72)

Reply:
We can simply prove this, by either asking first for the name, or by searching the source code of grk3, if that function (`g_set_prgname`) is used somewhere. Also, the documentation for it mentions this: `If you are using GApplication the program name is set in g_application_run(). In case of GDK or GTK+ it is set in gdk_init(), which is called by gtk_init() and the GtkApplication::startup handler. The program name is found by taking the last component of argv[0].`.

So my question here is, why is this function not called, does it only apply to gtk4 or did we miss something else? Or is it set to xournalpp since it is the name of the executable?

Reply:
Indeed, it is set to xournalpp once `g_application_run()` is called (I tested it in `XournalMain::on_startup()`).
With the line `g_set_prgname("com.github.xournalpp.xournalpp");`, the name is set properly, and not reset by `g_application_run()`.

Reply:
I also found this in https://gitlab.gnome.org/GNOME/gtk/-/blob/main/docs/reference/gtk/migrating-3to4.md

> Set a proper application ID
In GTK 4 we want the application's GApplication 'application-id' (and
therefore the D-Bus name), the desktop file basename and Wayland's xdg-shell
app_id to match. In order to achieve this with GTK 3.x call
g_set_prgname() with the same application ID you passed to
[[class@Gtk.Application](mailto:class@Gtk.Application)].  Rename your desktop files to match the application
ID if needed.
The call to g_set_prgname() can be removed once you fully migrated to GTK 4.
You should be aware that changing the application ID makes your
application appear as a new, different app to application installers.
You should consult the appstream documentation for best practices
around renaming applications. 

So I think this PR is correct.